crates/storage/src/interface.rs

```diff
@@ -696,6 +706,31 @@ impl<S: StorageAdaptor> MainInterface<S> {
         Ok(actions)
     }
 
+    /// Compares a foreign entity with a local one, and applies the resulting
+    /// actions to bring the two entities into sync.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if there are issues accessing local
+    /// data or if there are problems during the comparison process.
+    ///
+    pub fn compare_affective<D: Data>(
+        data: Option<Vec<u8>>,
+        comparison_data: ComparisonData,
+    ) -> Result<(), StorageError> {
+        let (local, remote) = Interface::compare_trees::<D>(data, &comparison_data)?;
+
+        for action in local {
```

Comment:
why not use map here?

Reply:
care to elaborate? how would we use map?

Reply:
actually local.for_each(...) is what i wanted to refer to.